build-tools/packages/version-tools/src/semver.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as semver from "semver";
+import { adjustVersion, VersionBumpType } from "@fluidframework/build-tools";
+import { bumpInternalVersion, getVersionRange } from "./internalVersionScheme";
+import { detectVersionScheme } from "./schemes";
+
+/**
+ * Return the version RANGE incremented by the bump type (major, minor, or patch).
+ *
+ * @remarks
+ *
+ * Only simple "^" and "~" ranges and Fluid internal version scheme ranges are supported.
+ *
+ * @param range - A dependency range string to increment.
+ * @param bumpType - The type of bump.
+ * @param prerelease - If true, will bump to a prerelease version.
+ * @returns a bumped range string.
+ */
+export function incRange(range: string, bumpType: VersionBumpType, prerelease = false): string {
+    if (semver.validRange(range) === null) {
+        throw new Error(`${range} is not a valid semver range.`);
+    }
+
+    const scheme = detectVersionScheme(range);
+    switch (scheme) {
+        default:
```

Comment:
We don't ever expect the default case to be hit, right? Should we throw an error in the default case then? So we catch issues if/when changes are made to `detectVersionScheme`?

Reply:
Good idea. Will do.